FileTypeSupport.MXLIFF/MXLIFFWriter.cs

```diff
@@ -33,109 +32,107 @@ public void Dispose()
         {
-            originalFileProperties = fileProperties;
         }
 
         public void Initialize(IDocumentProperties documentInfo)
         {
-            textExtractor = new MXLIFFTextExtractor();
+	        _documentProperties = documentInfo;
+            _textExtractor = new MXLIFFTextExtractor();
         }
 
         public void ProcessParagraphUnit(IParagraphUnit paragraphUnit)
         {
-            string id = paragraphUnit.Properties.Contexts.Contexts[0].GetMetaData("ID");
-            XmlNode xmlUnit = targetFile.SelectSingleNode("//x:trans-unit[@id='" + id + "']", nsmgr);
+            var id = paragraphUnit.Properties.Contexts.Contexts[0].GetMetaData("ID");
+            var xmlUnit = _targetFile.SelectSingleNode("//x:trans-unit[@id='" + id + "']", _nsmgr);
 
             CreateParagraphUnit(paragraphUnit, xmlUnit);
         }
 
         public void SetFileProperties(IFileProperties fileInfo)
         {
-            targetFile = new XmlDocument();
-            targetFile.PreserveWhitespace = false;
-            targetFile.Load(originalFileProperties.OriginalFilePath);
-            nsmgr = new XmlNamespaceManager(targetFile.NameTable);
-            nsmgr.AddNamespace("x", "urn:oasis:names:tc:xliff:document:1.2");
-            nsmgr.AddNamespace("m", "http://www.memsource.com/mxlf/2.0");
-
-            var level = targetFile.DocumentElement.Attributes["m:level"];
-
-            if (level != null)
-            {
-                workflowLevel = Int32.Parse(level.Value);
-            }
+	        _targetFile = new XmlDocument
+	        {
+		        PreserveWhitespace = false
+	        };
+	        var lastSavedPath = _documentProperties.LastSavedAsPath;
+	        var lastOpenPath = _documentProperties.LastOpenedAsPath;
+	        if (!string.IsNullOrEmpty(lastSavedPath))
+	        {
+				_targetFile.Load(lastSavedPath);
+			}else if (!string.IsNullOrEmpty(lastOpenPath))
+	        {
+				_targetFile.Load(lastOpenPath);
+			}
+			_nsmgr = new XmlNamespaceManager(_targetFile.NameTable);
+            _nsmgr.AddNamespace("x", "urn:oasis:names:tc:xliff:document:1.2");
+            _nsmgr.AddNamespace("m", "http://www.memsource.com/mxlf/2.0");
 
             // Acquire users
-            var memsourceUsers = targetFile.SelectNodes("//m:user", nsmgr);
-
-            if (memsourceUsers != null)
-            {
-                foreach (XmlElement user in memsourceUsers)
-                {
-                    var id = user.Attributes["id"]?.Value;
-                    var username = user.Attributes["username"]?.Value;
-
-                    if (!string.IsNullOrWhiteSpace(username) && !string.IsNullOrWhiteSpace(id) && users.ContainsKey(username))
-                    {
-                        users.Add(username, id);
-                    }
-                }
-            }
+            var memsourceUsers = _targetFile.SelectNodes("//m:user", _nsmgr);
```

Comment:
Is it possible that _targetFile will be null?

Reply:
Is instanciated in line 63.